invoices/invoiceregistry.go

```diff
@@ -1732,3 +1705,39 @@ func (i *InvoiceRegistry) HodlUnsubscribeAll(subscriber chan<- interface{}) {
 
 	delete(i.hodlReverseSubscriptions, subscriber)
 }
+
+// copySingleClients copies i.SingleInvoiceSubscription inside a lock. This is
+// usefule when we need to iterate the map to send notifications.
+func (i *InvoiceRegistry) copySingleClients() map[uint32]*SingleInvoiceSubscription {
+	i.Lock()
```

Comment:
Use `i.RLock()` and `defer i.RUnlock()`? Same in `copyClients()`.

Reply:
changed!